### PR TITLE
Make module dependency check recursive

### LINKED ISSUE #97812: Build stucks if some recursive dependencies isn't satisfied
### Tested versions

master, 4.3

### System information

Debian testing; SCons 4.5.2; Python 3.12.6

### Issue description

Run `scons module_jpg_enabled="no"` after which you will see following warning:
```
WARNING: Disabling 'basis_universal' module as the following dependencies are not satisfied: jpg 
```
After that the build process will stuck.

If you run `scons module_jpg_enabled="no" module_basis_universal_enabled="no"`, you will get the following warning:
```
WARNING: Disabling 'ktx' module as the following dependencies are not satisfied: basis_universal 
```
But after that build process won't stuck.

If you run `scons module_webp_enabled="no"` or `scons module_basis_universal_enabled="no"` you won't get any warnings

so I think that the problem is with recursive dependencies

### Steps to reproduce

N/A

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by ghost**
Fixes #97812

The `env.disabled_modules` and `methods.disable_module` weren't used anywhere so the first one was repurposed and the second just deleted

### CODE REVIEW

File: methods.py [hunk 1]
```diff
@@ -428,19 +424,21 @@ def module_check_dependencies(self, module):
...
-            missing_deps.append(dep)
-
-    if missing_deps != []:
-        print_warning(
-            "Disabling '{}' module as the following dependencies are not satisfied: {}".format(
-                module, ", ".join(missing_deps)
+        if opt not in self or not self[opt] or not module_check_dependencies(self, dep):
+            missing_deps.add(dep)
+
+    if missing_deps != set():
```
Comment (Repiteo) [inline methods.py:434]:
```suggestion
    if missing_deps:
```

**Review by Repiteo — APPROVED** [review]
The module logic in general could probably use a broad refactor; didn't realize just how messy it could get until I went to verify the changes

Nonetheless, the logic is sound and the changes work! Changing the container type should be fine, as I doubt anything is using that variable directly

**Comment by dustdfg** [comment]
> The module logic in general could probably use a broad refactor; didn't realize just how messy it could get until I went to verify the changes

I would say it about half of the SCons itself.... Anyway, I don't know what messiness you noticed but I've noticed following:

1. Disabling/enabling one module leads to rebuild all the files (by feelings it is half of the engine) which directly/transitively depend on `modules/modules_enabled.gen.h`

2. `module_check_dependencies` is already used not as it was meant:

```python
def module_check_dependencies(self, module):
    """
    ...
    Meant to be used in module `can_build` methods.
    ...
    """
```
It is used only inside main `SConstruct` so I assume there could be some other bugs

3. There is some module sorting logic. I assume it is about: if A depend on B so we should build B before A. That takes into account optional dependencies but lots of the modules don't declare their optional dependencies and we doesn't see sudden crashes. Examples of such modules: `navigation` optionally depends on `csg` and `gridmap`; `text_server_fb` optionally depends on `freetype`, `svg` and `msdfgen`. So one of three things:
    1. I misunderstand module sorting
    2. Sorting (and subsequently optional dependencies) is useless
    3. We have hole in the bottom of the boat but it surprisingly doesn't want to sink

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**